0000-catalog.md

```diff
@@ -0,0 +1,789 @@
+catalog_id and product_id. For example:
+
+```
+get_catalog_product(catalog_id, product_id) // hash(catalog_name), product_id
+set_catalog_product(catalog_id, product_id, catalog_product) //hash(catalog_name), product_id, catalog_product
+```
+
+### Catalog Product Addressing in the Merkle-Radix State System
+
+The Grid GS1 Catalog Product state address will be prefixed with the Grid
+namespace of "621dee", in addition to "02" (product namespace), "01" for (gs1
+product namespace), and the catalog_id.
+
+Therefore, all addresses starting with:
+
+```
+"621dee" + "02" + "01 + "cad4e781e797712" // catalog_id
+```
+
+are Grid GS1 Catalog Products and are expected to contain the standard
+attributes of a Grid GS1 Product as well as the extra PropertyValues that
+are defined in the catalog product schema. These PropertyValues are in
+addition to the expected PropertyValues of a [Grid GS1
+Product](https://github.com/hyperledger/grid-rfcs/blob/fbedec06d70b16492fea9f6b1e87146c5fc56771/0000-product.md).
+
+GTIN formats consist of 14-digit "numeric strings" which include some amount of
+internal "0" padding depending on the specific GTIN format (GTIN-12, GTIN-13, or 
+GTIN-14). After the 10-hex-characters that are consumed by the Grid namespace 
+prefix, the product namespace prefix, and GS1 namespace prefix. There are 
+15-hex-characters consumed by the catalog id, leaving 45 hex characters
+remaining in the address.
+
+The 11 to 13 digits of the GTIN can be left padded with 30 to 32-hex-character
+zeroes and right padded with 2-hex-character zeroes to accommodate potential
+future storage associated with the GS1 Product representation, for example:
+
+```
+"621dee" + "02" + "01" + "cad4e781e797712" + "00000000000000000000000000000"
++ 14-character "numeric string" product_id + "00" // product_id == GTIN
+```
+
+A full Catalog Product address using the example GTIN from
+https://www.gtin.info/ would therefore be:
+
+```
+"621dee0201cad4e781e797712000000000000000000000000000000001234560001200"
+```
+
+### Catalog Product Schema Definition
+
+This schema defines the additional attributes required of a grid product, such
+that it may be considered a "catalog product". Catalog products should be 
+grouped with other catalog products. Once grouped, the grid catalog can be 
+shared with other participants in a grid network.
+
+Any grid product added to a catalog (thus becoming a catalog product) will need
+to contain a reference fields to the original product (product_id, 
+product_namespace, and owner) and a catalog_id. A catalog product schema may 
+also be defined to require additional fields for that catalog_product. This 
+example schema enforces a _required_ `catalog_id` & `status` field, as 
+well as _optional_ `prices` & `return_policy` field.
+
+Example schema:
+
+```
+Schema(
+    name="Catalog Product",
+    description="A grid product that can be grouped with other products via
+grid catalog",
+    owner = "Target"
```

Comment:
xref above question on the naming of this attribute. Should it be `catalog_owner`?

Reply:
I changed the above to just `owner`. Though, the reference above was to the owner of a catalog (the entity that created it). This owner is the entity that creates a `catalog_product` by extending the definition of its existing Grid product.